### ISSUE TITLE
Xcode 14 Carthage "this SDK is not supported by the compiler"

### ISSUE THREAD

**Issue by RushanB** [firebase/firebase-ios-sdk#10251]
### Description

When trying to run `carthage update --use-xcframeworks --platform iOS` or `carthage bootstrap --use-xcframeworks --platform iOS` for my cartfile.

```
binary "https://dl.google.com/dl/firebase/ios/carthage/FirebaseAnalyticsBinary.json"
binary "https://dl.google.com/dl/firebase/ios/carthage/FirebaseAuthBinary.json"
binary "https://dl.google.com/dl/firebase/ios/carthage/FirebaseCrashlyticsBinary.json"
binary "https://dl.google.com/dl/firebase/ios/carthage/FirebaseDatabaseBinary.json"
binary "https://dl.google.com/dl/firebase/ios/carthage/FirebaseFirestoreBinary.json"
binary "https://dl.google.com/dl/firebase/ios/carthage/FirebaseInAppMessagingBinary.json"
binary "https://dl.google.com/dl/firebase/ios/carthage/FirebaseMessagingBinary.json"
binary "https://dl.google.com/dl/firebase/ios/carthage/FirebaseStorageBinary.json"
binary "https://dl.google.com/dl/firebase/ios/carthage/FirebaseProtobufBinary.json"
binary "https://dl.google.com/dl/firebase/ios/carthage/FirebaseRemoteConfigBinary.json"
```

I am seeing this following error on Xcode 14.0 (14A309).

<img width="1404" alt="image" src="https://user-images.githubusercontent.com/16970644/191885720-45bcf9f2-c857-41dd-b3f1-0af2ba4bd4bc.png">

### Reproducing the issue

run `carthage update --use-xcframeworks --platform iOS`

### Firebase SDK Version

0.0.0

### Xcode Version

14.0

### Installation Method

Carthage

### Firebase Product(s)

All

### Targeted Platforms

iOS

### Relevant Log Output

```shell
Failed to build module 'FirebaseStorage'; this SDK is not supported by the compiler (the SDK is built with 'Apple Swift version 5.6 (swiftlang-5.6.0.323.62 clang-1316.0.20.8)', while this compiler is 'Apple Swift version 5.7 (swiftlang-5.7.0.127.4 clang-1400.0.29.50)'). Please select a toolchain which matches the SDK.
```

### If using Swift Package Manager, the project's Package.resolved

<details>
<summary>Expand <code>Package.resolved</code> snippet</summary>
<br>

```json

Replace this line with the contents of your Package.resolved.

```

</details>

### If using CocoaPods, the project's Podfile.lock

<details>
<summary>Expand <code>Podfile.lock</code> snippet</summary>
<br>

```yml

Replace this line with the contents of your Podfile.lock!

```

</details>

**Comment by ndtoan75** [comment]
I has the same problem

**Comment by rizafran** [comment]
Hi @RushanB, I've successfully installed the SDK after running the `carthage update --use-xcframeworks --platform iOS` command. Could you provide a sample app that reproduces the issue?

**Comment by ndtoan75** [comment]
@rizafran Carthage was successfully installed the SDK but
When build the project - Xcode 14.0 (14A309). I am seeing this following error
![image](https://user-images.githubusercontent.com/61063957/192691359-5e81339d-f49a-4a91-90e3-288d7ab85b44.png)

**Comment by morganchen12** [comment]
Similar to #10140

**Comment by RushanB** [comment]
After updating to Xcode 14 (14A309) and running carthage update I had this error. I ended up changing the "Build Active Architectures Only" setting to No for each of the targets (and main target) which seemed to fix it, again it was only an issue on the sim. I believe it has something to do with the Apple Silicone changes.

**Comment by RushanB** [comment]
@rizafran  Seems I needed to add these 3 missing xcframeworks which were generated in Carthage/Build folder even though I did not specify them in the Cartfile:

- FirebaseAppCheckInterop.xcframework
- FirebaseAuthInterop.xcframework
- FirebaseStorageInternal.xcframework

After adding in these now I am getting a different error

`Undefined symbol: _OBJC_CLASS_$_FIRHeartbeatController`

![image](https://user-images.githubusercontent.com/16970644/194675980-5cc0c83b-721c-4501-83a4-53de18ea2b75.png)

**Comment by paulb777** [comment]
Make sure that `FirebaseCoreInternal.xcframework` is included.

**Comment by RushanB** [comment]
@rizafran I also don't think you would be able to reproduce this on your Mac unless you are using Apple Silicon or Xcode via Rosetta enable - I think this is causing some issues with the latest Xcode compiling the project for specific architectures.

**Comment by RushanB** [comment]
@paulb777 Would you look at that..after adding in those missing frameworks, seems everything is working as expected.

- FirebaseAppCheckInterop.xcframework
- FirebaseAuthInterop.xcframework
- FirebaseStorageInternal.xcframework
- FirebaseCoreInternal.xcframework

Thanks!!

**Comment by RushanB** [comment]
Maybe you can also update the documentation to say that we need these frameworks as it un clear as of today.

https://github.com/firebase/firebase-ios-sdk/blob/master/Carthage.md

**Comment by paulb777** [comment]
Do you have a suggestion for improvement?

The first bullet includes `Note that FirebaseAnalyticsBinary must always be included.`

**Comment by RushanB** [comment]
Second bullet point:
Run carthage update
This should be optimized for xcframeworks:
Run carthage update --use-xcframeworks

Third bullet point: 
Use Finder to open Carthage/Build/iOS
This should be 
Use Finder to open Carthage/Build

We can also add in a bullet point to add these missing frameworks for xcode 14+ on apple m1 macs:
- FirebaseAppCheckInterop.xcframework
- FirebaseAuthInterop.xcframework
- FirebaseStorageInternal.xcframework
- FirebaseCoreInternal.xcframework

Also not sure if you are aware but there is a bitcode deprecation starting from xcode 14:
Source: https://developer.apple.com/documentation/xcode-release-notes/xcode-14-release-notes

**Comment by paulb777** [comment]
Thanks!

- I believe `--use-xcframeworks` has no effect for the Firebase Carthage distribution since it consists of `xcframeworks`.
- Fixed the third bullet in #10327 
- The missing directories should be included in the folders of the requested Firebase products.
